lib/active_record/connection_adapters/postgis/postgis_database_tasks.rb

```diff
@@ -36,54 +36,54 @@ def create(master_established = false)
 
         # Override to use su_username and su_password
         def establish_master_connection
-          establish_connection(configuration.merge(
-            "database"           => "postgres",
-            "password"           => su_password,
-            "schema_search_path" => "public",
-            "username"           => su_username
+          establish_connection(configuration_hash.merge(
+            database:           "postgres",
+            password:           su_password,
+            schema_search_path: "public",
+            username:           su_username
           ))
         end
 
         def establish_su_connection
-          establish_connection(configuration.merge(
-            "password"           => su_password,
-            "schema_search_path" => "public",
-            "username"           => su_username
+          establish_connection(configuration_hash.merge(
+            password:           su_password,
+            schema_search_path: "public",
+            username:           su_username
           ))
         end
 
         def username
-          @username ||= configuration["username"]
+          @username ||= configuration_hash[:username]
         end
 
         def quoted_username
           @quoted_username ||= ::ActiveRecord::Base.connection.quote_column_name(username)
         end
 
         def password
-          @password ||= configuration["password"]
+          @password ||= configuration_hash[:password]
         end
 
         def su_username
-          @su_username ||= configuration["su_username"] || username
+          @su_username ||= configuration_hash[:su_username] || username
         end
 
         def su_password
-          @su_password ||= configuration["su_password"] || password
+          @su_password ||= configuration_hash[:su_password] || password
         end
 
         def has_su?
-          @has_su = configuration.include?("su_username") unless defined?(@has_su)
+          @has_su = configuration_hash.include?(:su_username) unless defined?(@has_su)
           @has_su
```

Comment:
I find this pattern a little bit clearer, but may not be that important

```suggestion
          @has_su if defined?(@has_su)
          
          @has_su = configuration_hash.include?(:su_username)
```

Reply:
Sorry for jumping in here. But I think, there's a bug in the suggested change. It should be

```diff
- @has_su if defined(@has_su)
+ return @has_su if defined(@has_su)
```

otherwise, the value for `@has_su` would be computed repeatedly.

Reply:
@schmidt of course, edited!